### ISSUE TITLE
go-yaml v1.14.3 panic on the the input JSON string for the map slice type. 

### ISSUE THREAD

**Issue by Peefy** [goccy/go-yaml#543]
**Describe the bug**

**To Reproduce**

Please provide a minimum yaml content that can be reproduced.
We are more than happy to use [Go Playground](https://go.dev/play)

Use the go 1.23 and `github.com/goccy/go-yaml v1.14.3` to run the following test, it will panic with the error message but `github.com/goccy/go-yaml` `v1.13.8` work fine

```go
func TestPanicJsonYamlMapSlice(t *testing.T) {
	code := `
{
	"apiVersion": "apps/v1",
	"kind": "Deployment",
	"metadata": {
		"name": "nginx-deployment",
		"labels": {
			"app": "nginx"
		}
	},
	"spec": {
		"replicas": 3,
		"selector": {
			"matchLabels": {
				"app": "nginx"
			}
		},
		"template": {
			"metadata": {
				"labels": {
					"app": "nginx"
				}
			},
			"spec": {
				"containers": [
					{
						"name": "nginx-container",
						"image": "nginx:latest",
						"ports": [
							{
								"containerPort": 80
							}
						]
					}
				]
			}
		}
	}
}	
`
	yamlData := &yaml.MapSlice{}
	if err := yaml.UnmarshalWithOptions([]byte(code), yamlData, yaml.UseOrderedMap(), yaml.UseJSONUnmarshaler()); err != nil {
		panic(err)
	}
}
```

The error message is

```txt
panic: [2:1] found character '  ' that cannot start any token
           1 | {
        >  2 |  "apiVersion": "apps/v1",
               ^
           3 |  "kind": "Deployment",
           4 |  "metadata": {
           5 |          "name": "nginx-deployment", [recovered]
        panic: [2:1] found character '  ' that cannot start any token
           1 | {
        >  2 |  "apiVersion": "apps/v1",
               ^
           3 |  "kind": "Deployment",
           4 |  "metadata": {
           5 |          "name": "nginx-deployment",

```

**Expected behavior**

Test passed

**Screenshots**

**Version Variables**
 - Go version: [e.g. 1.21 ] 1.23
 - go-yaml's Version: [e.g. v1.11.1 ] v1.14.3

**Additional context**

**Comment by goccy** [comment]
@Peefy  Probably because of the use of the tab character. This appears to be the correct behavior, since YAML does not allow tabs to be specified for indent. For example, can `go-yaml/yaml` parse it ?

**Comment by goccy** [comment]
Thank you for your additional information.
I see. It’s likely that in flow mode, allow tab characters for indentation—except in literal/folded styles—might be the correct behavior.

**Comment by goccy** [comment]
@Peefy  I've fixed this problem with v1.15.0 ! Please confirm it 🙏
